server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchFieldsContext.java

```diff
@@ -25,12 +25,18 @@
  */
 public class FetchFieldsContext {
     private final List<FieldAndFormat> fields;
+    private final boolean includeUnmapped;
 
-    public FetchFieldsContext(List<FieldAndFormat> fields) {
+    public FetchFieldsContext(List<FieldAndFormat> fields, boolean includeUnmapped) {
```

Comment:
Something to think about: if we end up making this configurable through a flag like `include_unmapped`, we could introduce a top-level parameter as we do here, or instead support the flag alongside each feield pattern (so it'd be part of `FieldAndFormat`).